libs/asset-viewer/src/lib/state/asset-search/asset-search.effects.ts

```diff
@@ -0,0 +1,296 @@
+import { Injectable, inject } from '@angular/core';
+import { ActivatedRoute, Params, Router } from '@angular/router';
+import { Actions, concatLatestFrom, createEffect, ofType } from '@ngrx/effects';
+import { Store } from '@ngrx/store';
+import { filter, map, switchMap, tap, withLatestFrom } from 'rxjs';
+
+import { appSharedStateActions } from '@asset-sg/client-shared';
+import { isNotNull } from '@asset-sg/core';
+import { AssetSearchQuery, AssetSearchResult, LV95, Polygon } from '@asset-sg/shared';
+
+import { AssetSearchService } from '../../services/asset-search.service';
+
+import * as actions from './asset-search.actions';
+import { AppStateWithAssetSearch, AssetSearchState } from './asset-search.reducer';
+import { selectAssetSearchQuery, selectAssetSearchState } from './asset-search.selector';
+
+
+@Injectable()
+export class AssetSearchEffects {
+  private readonly store = inject(Store<AssetSearchState>);
+  private readonly actions$ = inject(Actions);
+  private readonly router = inject(Router);
+  private readonly route = inject(ActivatedRoute);
+  private readonly assetSearchService = inject(AssetSearchService);
+  private readonly searchStore = inject(Store<AppStateWithAssetSearch>);
+
+
+  // noinspection JSUnusedGlobalSymbols
+  $readSearchQueryParams = createEffect(() => this.actions$.pipe(
+      ofType(actions.readParams),
+      concatLatestFrom(() => this.route.queryParams),
+      map(([_, params]) => {
+        const query: AssetSearchQuery = {};
+        const assetId: number | undefined = readNumberParam(params, QUERY_PARAM_MAPPING.assetId);
+        query.text = readStringParam(params, QUERY_PARAM_MAPPING.text);
+        query.polygon = readPolygonParam(params, QUERY_PARAM_MAPPING.polygon);
+        query.authorId = readNumberParam(params, QUERY_PARAM_MAPPING.authorId);
+        const min = readDateParam(params, QUERY_PARAM_MAPPING.createDate.min);
+        const max = readDateParam(params, QUERY_PARAM_MAPPING.createDate.max);
+        query.createDate = min && max ? { min, max } : undefined;
+        query.manCatLabelItemCodes = readArrayParam(params, QUERY_PARAM_MAPPING.manCatLabelItemCodes);
+        query.assetKindItemCodes = readArrayParam(params, QUERY_PARAM_MAPPING.manCatLabelItemCodes);
+        query.usageCodes = readArrayParam(params, QUERY_PARAM_MAPPING.usageCodes);
+        query.geomCodes = readArrayParam(params, QUERY_PARAM_MAPPING.geomCodes);
+        query.languageItemCodes = readArrayParam(params, QUERY_PARAM_MAPPING.languageItemCodes);
+        return { query, assetId };
+      }),
+      filter(({ query }) => Object.values(query).some(value => value !== undefined)),
+      map(({ query }) => {
+        this.store.dispatch(actions.searchByFilterConfiguration({ filterConfiguration: query }));
+        return actions.getStats();
+      }),
+    ),
+  );
+
+  $readAssetIdQueryParam = createEffect(() => this.actions$.pipe(
```

Comment:
Move the `$` to the right for all effects.
```suggestion
  readAssetIdQueryParam$ = createEffect(() => this.actions$.pipe(
```